docs/DeveloperGuide.md

```diff
@@ -640,84 +639,40 @@ Guarantees: Viewing of application is successful
   1b1. InternHunter displays an error message and informs the user that the index is out of bounds. <br />
   Use case resumes from step 1.
 
-#### Use case: UC12 - Add user profile item
+#### Use case: UC14 - Finding applications
 
-Guarantees: Addition of user profile item is successful
+* Similar to finding companies except user is finding applications.
 
-**MSS**
+#### Use case: UC15 - List all applications
 
-1.  User requests to add a user profile item to the user profile and provides details.
-2.  InternHunter adds the user profile item to the user profile. <br />
-    Use case ends.
+* Similar to listing all companies except user is listing all applications
 
-**Extensions**
+#### Use case: UC16 - Add user profile item
 
- 1a. InternHunter detects an error in the input format. <br />
-  1a1. InternHunter displays an error message and informs the user of the valid input format. <br />
-  Use case resumes from step 1.
+* Similar to adding a company except user is adding a user profile item.
 
-#### Use case: UC13 - Delete a user profile item 
+#### Use case: UC15 - Delete a user profile item 
 
-Precondition: User already has an existing list of user profile items <br/>
-Guarantees: Deletion of user profile item is successful
+* Similar to deleting a company except user is deleting a user profile item.
 
-**MSS**
+#### Use case: UC16 - Edit a user profile item
 
-1.  User requests to delete a user profile item from the user profile.
-2.  InternHunter removes the user profile item from the user profile. <br/>
-    Use case ends.
+* Similar to editing a company except user is editing a user profile item.
 
-**Extensions**
-
- 1a. InternHunter detects an error in the input format. <br/>
-  1a1. InternHunter displays an error message and informs the user of the valid input format. <br/>
-  Use case resumes from step 1.
-
- 1b. InternHunter detects an invalid index. <br/>
-  1b1. InternHunter displays an error message and informs the user that the index is out of bounds. <br/>
-  Use case resumes from step 1.
+#### Use case: UC17 - View a user profile item
 
-#### Use case: UC14 - Edit a user profile item
+* Similar to viewing a company except  user is viewing a user profile item.
 
-Precondition: User already has an existing list of user profile items <br/>
-Guarantees: Editing of user profile item is successful
+#### Use case: UC17 - Finding user profile items
```

Comment:
Numbering issue, there are 2 UC17

Reply:
> > > I just realised, but just for plausible consideration, should we add-in extension for deleting companies that indicate internships being deleted as well as its application.
> > 
> > 
> > Good point! Not sure how to appropriately frame it and if it should even be an extension of company or part of internship/application. @shawn-nyk @keanecjy What do you guys think?
> 
> I think we can just put it in the MSS, something like deletes the company and associating internships and applications along with it. This way means we will have more unique cases so it doesn't feel so repetitive too

But doing this you cannot use the reuse the UC for the other item types